src/realty/domain/realty-images.ts

```diff
@@ -0,0 +1,35 @@
+import { RealtyImagesErrorCodes } from './realty-images-error-codes';
+import { RealtyImagesException } from './realty-images.exception';
+
+const MAX_IMAGES = 5;
+
+interface RealtyImagesProps {
+  url: string;
+  isCover: boolean;
+}
+
+export class RealtyImages {
+  private images: RealtyImagesProps[] = [];
+
+  public setImage(image: RealtyImagesProps): void {
+    if (!image.url) {
+      throw new RealtyImagesException(RealtyImagesErrorCodes.IMAGE_IS_REQUIRED);
+    }
+
+    this.checkMaxImagesLimit();
```

Comment:
Acredito que seja uma boa prática evitar que um método seja chamado apenas para dar ou não uma excessão. Fica mais difícil para quem está lendo o código entender as regras. Se quiser extrair, me parece mais interessante extrair apenas a comparação e manter o comportamento aqui.
```suggestion
    if (this.hasReachedMaxImagesLimit()) {
      throw new RealtyImagesException(
        RealtyImagesErrorCodes.MAX_IMAGES_EXCEEDED,
      );
    }
```

Reply:
Bom ponto, de fato fica mais semântico dessa forma. Adicionei a resolução em 9dcd283c921cd33423fd5dd7e6500ea37cb5f6bf